File: Tienda_JAVA/src/javaapplication45/Limpieza.java

package javaapplication45;

public class Limpieza extends Producto{
    private static int ID = 100;
    int id;
    String marca;
    
    public Limpieza ( String nombre, double precio, String fechaImportacion, int stock, String marca){
        super( nombre,  precio,  fechaImportacion,  stock);
        this.id = ID;
        this.marca=marca;
        ID++;
        
    }

    public int getId() {
        return id;
    }
    
    public String ObtenerInformaciopnconMarca(){
    return super.obtenerInformacion() +", marca "+ marca+ "ID: "+id;
}
}

File: Tienda_Pyton/Limpieza.py
from Producto import Producto
class Limpieza(Producto):
    ID = 100

    def __init__(self, nombre=None, precio=None, fecha_importacion=None, stock=None, marca=None):
        super().__init__(nombre, precio, fecha_importacion, stock)
        self.id = Limpieza.ID
        self.marca = marca
        Limpieza.ID += 1

    def get_id(self):
        return self.id

    def obtener_informacion_con_marca(self):
        return super().obtener_informacion() + ", marca " + self.marca + " ID: " + str(self.id)
